### ISSUE TITLE
`clippy::exit` triggers, but expect claims that it doesn't

### ISSUE THREAD

**Issue by BenjaminBrienen** [rust-lang/rust-clippy#15107]
### Summary

If I expect the lint, then it claims it doesn't trigger

### Reproducer

I tried this code:

```rust
#[expect(clippy::exit, reason = "TODO: use clap")] // this lint expectation is unfulfilled
fn main() {
    std::process::exit(1);
}
```

```rust
fn main() {
    std::process::exit(1); // usage of `process::exit`
}
```

### Version

```text

```

**Comment by BenjaminBrienen** [comment]
I feel like I'm losing my mind.

```rs
#![expect(unfulfilled_lint_expectations)] // this lint expectation is unfulfilled

#[expect(clippy::exit, reason = "TODO: use clap")] // this lint expectation is unfulfilled
fn main() {
    std::process::exit(1);
}
```

**Comment by profetia** [comment]
If you use `expect`, it will then warn if the lint is NOT triggered.

**Comment by BenjaminBrienen** [comment]
The expect is warning that the lint is not triggered. It should be triggered, though. If the expect is not there, then the lint does trigger!

**Comment by blyxyas** [comment]
>```
>#![expect(unfulfilled_lint_expectations)] // this lint expectation is unfulfilled
>#[expect(clippy::exit, reason = "TODO: use clap")] // this lint expectation is unfulfilled
>fn main() {
>    std::process::exit(1);
>}
>```

What you mean is probably `#![deny(unfulfilled_lint_expectations)]` not `#![expect(unfulfilled_lint_expectations)]`

Apart from that, the lint is behaving as expected, we don't lint for that lint on `fn main` because it's expected that `fn main` can and will exit eventually.

If you `expect` that the lint will be triggered, that `expect` will fail because we don't check for that lint in `main`.

>  If the expect is not there, then the lint does trigger!

I'm not seeing that behaviour with the latest stable Clippy. If you use this:

```
#[deny(clippy::exit, reason = "TODO: use clap")]
fn main() {
    std::process::exit(1);
}
```

It won't trigger because as I said, we don't check for `clippy::exit` on `main`.

**Comment by BenjaminBrienen** [comment]
I now understand that it doesn't trigger on main. I think the first expect in this screenshot is still a bug, though.

![Image](https://github.com/user-attachments/assets/ea4002d3-b6c1-4cf9-80af-cfa2a41c702b)

I could have sworn that I was originally having the problem in a function that was not main, but now the expect seems to be working properly there.
